### ISSUE TITLE
Support  `selection` as parameter for `replaceText` - so we can transform selected text from smart fragments

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/SmartTemplates#289]
Usage:

We want to do replacements in the selected text, for instance, the string

`Menu >> submenu >> command `

should be transformed into

`Menu » submenu » command `

Additionally we could add some markup to display the text in italic (or wrap a layout span around it). Example for such a smart fragment:

```
<i>%replaceText(">>","»",selection)%</i>
```

Ideally the text (html) should still be selected after the replacement operation, this way we could concatenate multiple such commands for some powerful formatting.

**Comment by RealRaven2000** [comment]
Here is new version with the funciton added. Result:

![image](https://github.com/RealRaven2000/SmartTemplates/assets/6314674/e99f5585-55b4-4416-81a8-3d6821a29653)

By highlighting a line below (even if it has additional HTML such as bold or colored) we can transform it to a cursive string replacing all instances of ">>" with the Guillemet (right pointing double angle quotation mark) by using the following fragment:

```
<i>%replaceText(">>","»",selection)%</i>
```

At the moment if removes the existing selection, so re-selecting the area would be a nice-to-have. But this is already very useful to me!

[smartTemplate-fx-4.5pre42.zip](https://github.com/RealRaven2000/SmartTemplates/files/15148329/smartTemplate-fx-4.5pre42.zip)

-------

To install, download archive and drop into Thunderbird Add-ns Manager (don't extract contents).

**Comment by RealRaven2000** [comment]
Found some regressions that made `%matchTextFromBody()%` misbehave. Fixed in the version below:

[smartTemplate-fx-4.5pre44.zip](https://github.com/RealRaven2000/SmartTemplates/files/15151436/smartTemplate-fx-4.5pre44.zip)

-------

To install, download archive and drop into Thunderbird Add-ns Manager (don't extract contents).

**Comment by RealRaven2000** [comment]
Added documentation in Variables tab + localized to most supported languages:

[smartTemplate-fx-4.5pre56.zip](https://github.com/RealRaven2000/SmartTemplates/files/15153674/smartTemplate-fx-4.5pre56.zip)

-----

To install, download archive and drop into Thunderbird Add-ns Manager (don't extract contents).

**Comment by RealRaven2000** [comment]
Found another regression that had problems with multiple smartTempalte variables in the same line (overmatching and combining multiple lines) such as `To: "%to(name)%"&lt;%to("( %",mail)%&gt;`

[smartTemplate-fx-4.5pre57.zip](https://github.com/RealRaven2000/SmartTemplates/files/15175563/smartTemplate-fx-4.5pre57.zip)

this one should be more stable

-----

To install, download archive and drop into Thunderbird Add-ns Manager (don't extract contents).

**Comment by RealRaven2000** [comment]
Implemented in 4.5 released on 14/05/2024
